pandas/core/base.py

```diff
@@ -677,19 +677,25 @@ def item(self):
         """
         Return the first element of the underlying data as a python scalar.
 
-        .. deprecated:: 0.25.0
-
         Returns
         -------
         scalar
             The first element of %(klass)s.
+
+        Raises
+        ------
+        ValueError
+            If the data is not length-1.
         """
-        warnings.warn(
-            "`item` has been deprecated and will be removed in a future version",
-            FutureWarning,
-            stacklevel=2,
-        )
-        return self.values.item()
+        if not needs_i8_conversion(self.dtype):
+            # numpy returns ints instead of datetime64/timedelta64 objects,
+            #  which we need to wrap in Timestamp/Timedelta/Period regardless.
+            return self.values.item()
```

Comment:
This won't work for ExtensionArrays. We can discuss adding `item` to the interface, but I would rather (or at least for now) let ExtensionArrays take the path you have below that uses iteration (which should already handle the conversion to a python scalar)

Reply:
I would be +1 on adding to EA arrays, why have inconsistency in code paths.

Reply:
> This won't work for ExtensionArrays

This uses `.values`, so will convert to ndarray and then call `item`.  So it shouldn't be any more broken than what we have now.

Reply:
i have no objection to adding item to EAs separately

Reply:
> So it shouldn't be any more broken than what we have now.

And to fix that, the only thing that is needed is adding a `and not is_extension_array_dtype(self.dtype):` to the above `if` check.

Reply:
im happy to do that here, will need tests in a follow-up